apps/gallery/js/info.js

```diff
@@ -44,18 +52,72 @@ function showFileInformation(fileinfo) {
           args: args
         },
         'info-type': {raw: fileinfo.type},
-        'info-date': {raw: MediaUtils.formatDate(fileinfo.date)},
+        'info-date': {raw: formattedDate},
         'info-resolution': {
           raw: fileinfo.metadata.width + 'x' + fileinfo.metadata.height
-        }
+        },
       };
+      // placeholders for Exif
+      exifFields.forEach((elem) => {
+        data[elem] = {raw: ''};
+      });
 
       // Populate info overlay view
       MediaUtils.populateMediaInfo(data);
+
+      LazyLoader.load(['shared/js/media/jpeg-exif.js'], () => {
+        photodb.getFile(fileinfo.name, (imagefile) => {
+          JPEGParser.readExifMetaData(imagefile, (error, metadata) => {
+            if (!metadata) {
+              // hide the exif fields.
+              exifFields.forEach((elem) => {
+                setFieldVisibility(elem, false);
```

Comment:
Maybe start off with these hidden even before waiting for the EXIF to be read?  Seems like there might be a flash of the fields being shown and then hidden again.